src/lp_sensitivity2.jl

```diff
@@ -0,0 +1,428 @@
+                    continue  # Direction is ≈0 so any value of δ is okay.
+                end
+                # There are three confusing sign switch opportunities. We
+                # usually want to be:
+                # - minimizing (switch if maximizing)
+                # - with d[j][e_i] ≥ 0 (switch if ≤ 0)
+                # - and nonbasic at the lower bound (switch if upper)
+                # If an odd number of these things is true, then the ratio
+                # forms an upper bound for δ. Otherwise, it forms a lower bound.
+                stat = j <= n ? basis.variables[j] : basis.constraints[j - n]
+                if xor(is_min, d[j][e_i] > atol, stat == MOI.NONBASIC_AT_LOWER)
+                    t_hi = min(t_hi, π[j] / d[j][e_i])
+                else
+                    t_lo = max(t_lo, π[j] / d[j][e_i])
+                end
+            end
+            report.objective[var] = (t_lo, t_hi)
+        end
+    end
+
+    return report
+end
+
+_basic_range(con, set::MOI.LessThan) = (value(con) - set.upper, Inf)
+_basic_range(con, set::MOI.GreaterThan) = (-Inf, value(con) - set.lower)
+_basic_range(con, set) = (0.0, 0.0)
+
+"""
+    _compute_rhs_range(d_B, x_B, l_B, u_B, atol)
+
+Assume we start with the optimal solution `x_old`, we want to compute a step
+size `t` in a direction `d` such that `x_new = x_old + t * d` is still
+represented by the same optimal basis. This can be computed a la primal simplex
+where we use an artificial entering variable.
+
+    A * x_new = A * (x_old + t * d)
+                = A * x_old + t * A * d
+                = 0         + t * A * d  # Since A * x_old = 0
+    =>  A * d = 0
+    => B * d_B + N * d_N = 0
+    => d_B = B \\ -(N * d_N)
+
+Note we only have to compute the basic component of the direction vector,
+because `d_N` is just zeros with a `1` in the component associated with the
+artificial entering variable. Therefore, all that remains is to compute the
+associated column of `N`.
+
+If we are increasing the bounds associated with the `i`th decision variable,
+then our artificial entering variable is a duplicate of the `i`th variable, and
+`N * d_N = A[:, i]`.
+
+If we are increasing the bounds associated with the `i`th affine constraint,
+then our artificial entering variable is a duplicate of the slack variable
+associated with the `i`th constraint, i.e., a `-1` in the `i`th row and zeros
+everywhere else.
+
+In either case:
+
+    d_B = -(B \\ A[:, i])
+
+Now, having computed a direction such that `x_new = x_old + t * d`. By ensuring
+that `A * d = 0`, we maintained structural feasibility. Now we need to compute
+bounds on `t` such that `x_new` maintains bound feasibility. That is, compute
+bounds on t such that:
+
+    l_B[j] <= x_B[j] + t * d_B[j] <= u_B[j].
+"""
+function _compute_rhs_range(d_B, x_B, l_B, u_B, atol)
+    t_lo, t_hi = -Inf, Inf
+    for j = 1:length(l_B)
+        if d_B[j] > atol
+            t_hi = min(t_hi, (u_B[j] - x_B[j]) / d_B[j])
```

Comment:
In other places in this function, you have `t_lo, t_hi`, but here you do `t_hi` first.  Also, the expressions in both branches are the same, so could compute them beforehand?

Reply:
I re-ordered the lines. But I didn't find computing the `t` any more readable. It just added another layer of in-direction.